source/lac/sparse_direct.cc

```diff
@@ -928,66 +928,175 @@ SparseDirectMUMPS::initialize_matrix(const Matrix &matrix)
 {
   Assert(matrix.n() == matrix.m(), ExcMessage("Matrix needs to be square."));
 
-  // Hand over matrix to MUMPS as centralized assembled matrix
-  if (Utilities::MPI::this_mpi_process(mpi_communicator) == 0)
+  n    = matrix.n();
+  id.n = n;
+
+  if constexpr (std::is_same_v<SparseMatrix<double>, Matrix>)
```

Comment:
People usually say things such as "if x is 4, then ...", rather than "if 4 is x, then ...". In other words, in "if" expressions, the *variable* usually appears on the left and the constant against which you compare on the right. Here, you turn this around by asking "if SparseMatrix is equal to Matrix". Would you mind using the other order?